### ISSUE TITLE
Cannot edit menu link content

### ISSUE THREAD

**Issue by timcosgrove** [department-of-veterans-affairs/va.gov-cms#7735]
## Describe the defect
Attempting to edit a menu link results in an error.

## To Reproduce
1. Visit https://prod.cms.va.gov/admin/structure/menu/manage/education-benefits-hub for example.
2. Attempt to edit GI Bill School comparison

## Expected behavior
You should see an edit form for making adjustments to the menu link.

## Screenshots 
![image.png](https://images.zenhubusercontent.com/614213ae32396d2bc1205abd/b307f064-9914-4fe6-b18c-4ba582a25943)

 ## Additional context
In looking at the error logs, workbench_menu_access.module attempts to call a function `accessCheck()` on the menu links access control handler to determine if a user should have access to edit a given menu link.
https://git.drupalcode.org/project/workbench_menu_access/-/blob/8.x-1.x/workbench_menu_access.module#L54

`accessCheck()` is not the correct method name to call on a class that extends EntityAccessControlHandler: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Entity%21EntityAccessControlHandler.php/function/EntityAccessControlHandler%3A%3AcheckAccess/9.2.x

It is very possible these checks have never worked. 

### CMS Team

Please leave only the team that will do this work selected.

- [x] `Platform CMS Team`
- [x] `Sitewide CMS Team`
  - [ ] `⭐️ Content ops`
  - [ ] `⭐️ CMS experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product support`
  - [ ] `⭐️ User support`

**Comment by swirtSJW** [comment]
I am not following this error.  I am able to edit links at https://prod.cms.va.gov/admin/structure/menu/manage/education-benefits-hub and I am able to edit links. 
![image](https://user-images.githubusercontent.com/5752113/150900131-77fa52f2-94a3-406f-84a3-71009fc96067.png)
Are you encountering this error as a specific user?

**Comment by swirtSJW** [comment]
Also if workbench menu access is calling the wrong method, shouldn't this  be an issue  with a patch, instead of overriding it locally?

**Comment by swirtSJW** [comment]
This actually seems something specific to https://prod.cms.va.gov/admin/structure/menu/item/686/  I can edit the others.
Makes me suspect that menus with $ in them are messed up
![image](https://user-images.githubusercontent.com/5752113/150900928-e21e3a29-42e9-498f-8f09-332f4ccd02ba.png)

**Comment by indytechcook** [comment]
It a bit more complicated then that.  The `accessCheck` is a public function on the `WorkbenchMenuLinkContentAccessControlHandler` class.  The workbench module assumes that it's access handler is always used which is a dangerous assumption. I ran into this problem when first looking at menu link access issues in graphql.  

Workbench seems to be incompatible with newer versions of Drupal access checks. 
 This is probably related to https://www.drupal.org/project/drupal/issues/3043321

From WorkbenchMenuLinkContentAccessControlHandler.
```
  /**
   * Public alias for checkAccess().
   */
  public function accessCheck(EntityInterface $entity, $operation, AccountInterface $account) {
    return $this->checkAccess($entity, $operation, $account);

  }
```

The reason it works on some links but not other is probably related to how the check is done .  For example, with a patch we added, menu link entity checks now use the linked entity for checks.  This change to using the linked entity would explain differences per menu link.  For example, if the `$600 Buy-Up program`  might point to an entity bundle which is controlled by Workbench while others are not.  Or perhaps the access check returns Allowed before it gets to the Workbench check.  This are more likely then a character.  I'm not saying the character isn't related, it's just not likely.

**Comment by swirtSJW** [comment]
> For example, if the $600 Buy-Up program might point to an entity bundle which is controlled by Workbench while others are not

All content types are managed by workbench access.

**Comment by swirtSJW** [comment]
> It a bit more complicated then that.

Or it could be exactly that simple.
Removed the $  on staging.   
![image](https://user-images.githubusercontent.com/5752113/151004745-74f668b4-3d39-4847-80ef-2f26485d1cfd.png)

and then I could edit the menu item no problem,
![image](https://user-images.githubusercontent.com/5752113/151004809-595cdb93-1c07-45a1-a204-a30770ea93d0.png)

**Comment by swirtSJW** [comment]
I think it is more related to this.  https://github.com/department-of-veterans-affairs/va.gov-cms/issues/6948
I was able to put the $ back in and save and have it continue to work.   I think it stems from the way we are injecting the node title into the menu title.  I think there is some sanitization missing that is allowing bad chars in.

**Comment by swirtSJW** [comment]
I think this error is misleading.
```
Error: Call to undefined method Drupal\va_gov_graphql\Access\MenuLinkContentAccessHandler::accessCheck() in workbench_menu_access_alter_menu_link_forms() (line 54 of /var/www/cms/docroot/modules/contrib/workbench_menu_access/workbench_menu_access.module)
```
I think it is not because method does not exist... if that were true we would see it on every menu load (thousands of times)  I think it errors in this case because it could not properly load that particular menu so none of the methods for the item 
 exist, because it had bad data and could not be loaded.
I just resaved https://prod.cms.va.gov/education/about-gi-bill-benefits/montgomery-active-duty/buy-up which cause a save to the menu title and there is no more error.
Workbench menu access is performing as expected... as long as we don't feed it a bad menu item.

**Comment by swirtSJW** [comment]
> It's not just $. The particular link I first saw this on was GI Bill school comparison which has no special characters.
Different failure.  That one fails because the "page" that it links to does not exist.

**Comment by swirtSJW** [comment]
Actually I suspect a similar failure on that one.   Notice the special char on the page the menu item should be pointing to
![image](https://user-images.githubusercontent.com/5752113/151028082-e347b543-d762-4b4b-bc02-a4b86d59d0cf.png)

**Comment by swirtSJW** [comment]
Tim it seems like most of the ones that mention GI BIll ____ are broken and my  money is on that someone created them with the registered symbol in them all.

Usually I can get it edit the name by opening the node and saving the node.  The problem is all the GI bill ones don't have a node.  We may have to fix them by deleting them and adding them back by hand.

**Comment by ndouglas** [comment]
I'm fairly confident after my work on #6948 today that this is another symptom.  

As Neil mentioned above:

>The reason it works on some links but not other is probably related to how the check is done . For example, with a patch we added, menu link entity checks now use the linked entity for checks. This change to using the linked entity would explain differences per menu link. For example, if the $600 Buy-Up program might point to an entity bundle which is controlled by Workbench while others are not. Or perhaps the access check returns Allowed before it gets to the Workbench check. These are more likely then a character. I'm not saying the character isn't related, it's just not likely.

I think the reason this code path is exercised (and fails) is because the "Save and Continue" code doesn't permit the `menu_item_extras` submit handler to run and fully populate the details of the menu item prior to its creation.

**Comment by ndouglas** [comment]
This should be closed by #7868.  I'll close this issue unless/until it reappears.
